earth2studio/data/xr.py

```diff
@@ -62,8 +62,117 @@ def __call__(
         return self.da.sel(time=time, variable=variable)
 
 
+class InferenceOuputSource:
+    """Adapt a inference output into a data source.
+
+    This data source loads an existing xarray Dataset, such as a NetCDF file or Zarr
+    store from an Earth2Studio forecast inference pipeline, which can then be filtered
+    to provide a data array give a variable and time.
+    Time, lead_time and variable are expected dimensions to be present in the DataSet.
+
+    Note
+    ----
+    This data source performs automatic transformation of time coordinates based on the
+    time and lead_time coordinates:
+
+    - **If lead_time has length 1**: The single lead_time value is added to all time
+      coordinates to produce valid forecast times (time + lead_time). The lead_time
+      dimension is then removed.
+    - **If time has length 1**: The single time value is broadcast to match the
+      lead_time dimension, then lead_time values are added to produce valid forecast
+      times. The time dimension is removed and lead_time is renamed to time.
+
+    Either time or lead_time must have length 1 after applying the filter_dict
+    - both cannot have length > 1 simultaneously. The resulting dataset will have a
+    single time dimension containing the computed valid forecast timestamps.
+
+    Parameters
+    ----------
+    file_path : str
+        Path to an xarray-compatible dataset file (e.g., NetCDF/Zarr).
+    filter_dict : dict, optional
+        Dictionary of selections applied before transformation (e.g.
+        ``{"ensemble": 0}``). Coordinates not in the required
+        dimensions are dropped after selection, by default None
+    **xr_args : Any
+        Additional keyword arguments forwarded to ``xarray.open_dataset``.
+    """
+
+    def __init__(self, file_path: str, filter_dict: dict = None, **xr_args: Any):
+        self.file_path = file_path
+        self.da = xr.open_dataset(self.file_path, **xr_args)
+        self.da = self.da.to_array("variable")
+
+        # The following dimensions and their order is required for the data to be used as a datasource
+        required_dims = {"time", "lead_time", "variable"}
+        if filter_dict:
+            # Need to keep these dims, so make then a list if scalar value
+            for k in ("time", "lead_time"):
```

Comment:
Don't you want to check for variable filters here too maybe? Like ```for k in ("time", "lead_time", "variable"):```.
Also seems like the if statement bellow could be a bit less open ended but its fine.

Reply:
Nah, its just the time ones we need to make a list form to the dims dont get removed, probably a slicker way to do this but oh well.

Did clean up one if statement though!